addons/danger/functions/fnc_leaderFlank.sqf

```diff
@@ -0,0 +1,149 @@
+#include "script_component.hpp"
+/*
+ * Author: nkenny
+ * Leader calls for extended aggresive manoeuvres towards buildings or location
+ *
+ * Arguments:
+ * 0: Group leader <OBJECT>
+ * 1: Group threat unit <OBJECT> or position <ARRAY>
+ * 2: Units in group, default all <ARRAY>
+ * 3: How many assault cycles, default four <NUMBER>
+ * 4: Overwatch destination <ARRAY>
+ *
+ * Return Value:
+ * success
+ *
+ * Example:
+ * [bob, angryJoe] call lambs_danger_fnc_leaderManoeuvre;
+ *
+ * Public: No
+*/
+params ["_unit", "_target", ["_units", []], ["_cycle", 3], ["_overwatch", []]];
+
+// stopped or static
+if (!(attackEnabled _unit) || {stopped _unit}) exitWith {false};
+
+// find target
+_target = _target call CBA_fnc_getPos;
+
+// check CQB ~ exit if in close combat other functions will do the work - nkenny
+if (_unit distance2D _target < GVAR(CQB_range)) exitWith {
+
+    [_unit, _target] call FUNC(leaderGarrison);
+
+    false
+};
+
+// find units
+if (_units isEqualTo []) then {
+    _units = [_unit, 200] call FUNC(findReadyUnits);
+};
+if (_units isEqualTo []) exitWith {false};
+
+// find vehicles
+private _vehicles = [];
+{
+    if (!(isNull objectParent _x) && { isTouchingGround vehicle _x } && { canFire vehicle _x }) then {
+        _vehicles pushBackUnique vehicle _x;
+    };
+} foreach (units _unit select { _unit distance _x < 350 && { canFire _x }});
+
+// sort building locations
+private _pos = [_target, 12, true, false] call FUNC(findBuildings);
+_pos pushBack _target;
+
+// find overwatch position
+if (_overwatch isEqualTo []) then {
+
+    _overwatch = selectBestPlaces [_target, ((_unit distance2D _target) / 2) min 200, "(2 * hills) + (2 * forest + trees + houses) - (2 * meadow) - (2 * windy) - (2 * sea) - (10 * deadBody)", 100 , 3] apply {[(_x select 0) distance2d _unit, _x select 0]};
+    _overwatch sort true;
+    _overwatch = _overwatch apply {_x select 1};
+    _overwatch = _overwatch select {!(surfaceIsWater _x)};
+    _overwatch pushBack ([getPos _unit, ((_unit distance2D _target) / 2) min 200, 100, 8, _target] call FUNC(findOverwatch));
```

Comment:
the distance calculation should be stored somewhere because in L58 is doning the same

Reply:
Fixed